A6lib.cpp

```diff
@@ -441,3 +441,31 @@ byte A6lib::A6waitFor(const char *resp1, const char *resp2, int timeout, String
     }
     return retVal;
 }
+
+
+// making modem functioning or free from previous engagements
+int A6lib::getFree(long timeout)
```

Comment:
1.  I will surely change the name and I too think FreeModem seems fine.
2.  The intention behind this is that in cases where the modem is still busy on last command or some byte got missed out for sending.  So modem is still expecting the inputs from the user.  Sending CR n LF or some combination of them to modem then a AT will ensure working of modem before issuing the next command otherwise reiterate over that.
I was thinking this as a measure to give robustness.  
Correct me here with your thoughts, If I am wrong or if this is not needed.